packages/kbn-dev-utils/src/run/run.ts

```diff
@@ -62,7 +62,8 @@ export async function run(fn: RunFn, options: Options = {}) {
 
   process.on('unhandledRejection', error => {
     log.error('UNHANDLED PROMISE REJECTION');
-    log.error(error);
+    if (error instanceof Error || typeof error === 'string') log.error(error);
+    else log.error(String(error));
```

Comment:
I'm not sure if this is actually what we want or if we want to relax the accepted types of `log.error()` instead... I'll leave that up to @elastic/kibana-operations 🙃

Reply:
Btw, the reason for this change because of this commit to `@types/node`:

https://github.com/DefinitelyTyped/DefinitelyTyped/commit/8266be6d7961bc497763b8fb26658cc16d6d8c87

Here the `UnhandledRejectionListener` goes from accepting `any` as the first argument to accepting `{} | null | undefined`

Reply:
Also, this type of `unhandledRejection` logic is [now part of Kibana core](https://github.com/elastic/kibana/commit/56010b1c6543cf6b3b2a8e39e415a7085296eabf) - so the whole function could be eliminated if we wanted to if I understand this use-case correctly

Reply:
Yeah, we should probably just require `src/setup_node_env/exit_on_warning.js` here rather than keeping both implementations. Either way you'll need to run `yarn kbn run build -i @kbn/pm` and commit the build output of `@kbn/pm` since it bundles `@kbn/dev-utils`

Reply:
Oh, I didn't think about the fact that it's outside of this package... we probably shouldn't violate the package boundaries for something like this.

Reply:
> we probably shouldn't violate the package boundaries for something like this.

Good point. Let's leave this `unhandledRejection` where it is then.

Are you ok with the modifications to the function then or do you have a simpler approach? I'm not super happy with it myself, but it was the best I could come up with without having to change the accepted arg type of `log.error()`.

Reply:
I prefer not to have single line if statements, but `log.error` is kinda picky about what it gets passed. That said, I think if the `error` argument isn't an error we should probably do a little more than stringify the value for the log. Maybe something like this?

```ts
import { inspect } from 'util';

// ...

log.error(
  error instanceof Error
    ? error
    : new Error(`non-Error type rejection value: ${inspect(error)}`)
)
```

Inspired by https://github.com/elastic/kibana/blob/master/packages/kbn-dev-utils/src/tooling_log/tooling_log_text_writer.ts#L56-L66

Reply:
Updated 👍